### PR TITLE
{bio}[foss/2021a] MAGeCK v0.5.9.4 w/ Python 3.9.5

### PR DESCRIPTION
**Pull request by branfosj**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u03a.bear.cluster - Linux RHEL 8.5, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/006dacfce28e9b037b0b3c6fbe17e817 for a full test report.

**Comment by casparvl** [comment]
Test report by @casparvl
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
software1.lisa.surfsara.nl - Linux debian 10.11, x86_64, Intel(R) Xeon(R) Silver 4110 CPU @ 2.10GHz, Python 2.7.16
See https://gist.github.com/85b66d2ad4a1b3bee86ca21df6e30ffd for a full test report.

**Comment by casparvl** [comment]
Hm, we run with `RPATH`. It seems that doesn't work:
```
== 2022-03-07 12:39:34,835 easyblock.py:2973 WARNING No '(RPATH)' found in 'readelf -d' output for /home/casparl/.local/easybuild/Debian10/2021/software/MAGeCK/0.5.9.4-foss-2021a/bin/RRA:
Dynamic section at offset 0xfdc0 contains 29 entries:
  Tag        Type                         Name/Value
 0x0000000000000001 (NEEDED)             Shared library: [libstdc++.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libm.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libgcc_s.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
....
== 2022-03-07 12:39:34,836 run.py:233 INFO running cmd: file /home/casparl/.local/easybuild/Debian10/2021/software/MAGeCK/0.5.9.4-foss-2021a/bin/mageckGSEA
== 2022-03-07 12:39:34,862 run.py:233 INFO running cmd: ldd /home/casparl/.local/easybuild/Debian10/2021/software/MAGeCK/0.5.9.4-foss-2021a/bin/mageckGSEA
== 2022-03-07 12:39:34,899 run.py:233 INFO running cmd: readelf -d /home/casparl/.local/easybuild/Debian10/2021/software/MAGeCK/0.5.9.4-foss-2021a/bin/mageckGSEA
== 2022-03-07 12:39:34,927 easyblock.py:2973 WARNING No '(RPATH)' found in 'readelf -d' output for /home/casparl/.local/easybuild/Debian10/2021/software/MAGeCK/0.5.9.4-foss-2021a/bin/mageckGSEA:
Dynamic section at offset 0x1fda0 contains 29 entries:
...
```
So, it's missing RPATH for the RRA and mageckGSEA binaries. @branfosj do you know of these binaries get compiled during the installation? Or is this essentially a binary installation? (some context to why RPATH doesn't work for binary installations with EasyBuild is [here](https://github.com/easybuilders/easybuild-framework/issues/3857))

Anyway, I don't think this should block the PR itself, since most users seem to use EasyBuild without RPATH anyway... A fix can always be implemented in a future PR, for now we've logged here the fact that it fails with `--rpath`.

I'll rerun without `rpath` support and see if that passes correctly on my system.

**Comment by casparvl** [comment]
Test report by @casparvl
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
software1.lisa.surfsara.nl - Linux debian 10.11, x86_64, Intel(R) Xeon(R) Silver 4110 CPU @ 2.10GHz, Python 2.7.16
See https://gist.github.com/135f1edf356c95374a08937433fa3523 for a full test report.

**Comment by branfosj** [comment]
I can see the issue. The binaries should be compiled during install - it is in the `setup.py` to do so. However, they ship the binaries already compiled and so they are not getting recompiled.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u03a.bear.cluster - Linux RHEL 8.5, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/50b6e164ec292344b7fb5c4984923348 for a full test report.

**Comment by branfosj** [comment]
@casparvl f9e95ca should fix the RPATH issue.

**Comment by casparvl** [comment]
Yes, all good now, also with `--rpath`. You found that issue quickly, kudo's! Since we're one of the few centers using `--rpath`, I know from experience it is sometimes not so easy to find out... (does a binary not get build but copied? Does it get build, but the EasyBuild RPATH wrappers somehow don't function properly? etc...)

File: easybuild/easyconfigs/m/MAGeCK/MAGeCK-0.5.9.4-foss-2021a.eb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+dependencies = [
+    ('Python', '3.9.5'),
+    ('SciPy-bundle', '2021.05'),
+]
+
+source_urls = [SOURCEFORGE_SOURCE]
+sources = [SOURCELOWER_TAR_GZ]
+checksums = ['42875b308f2d9ee38f083fb1020d7d48fca29e2100fdb09ced1969b738b8c939']
+
+# remove the shipped mageckGSEA and RRA binaries so that the Python install will build them
```
Comment (casparvl) [inline easybuild/easyconfigs/m/MAGeCK/MAGeCK-0.5.9.4-foss-2021a.eb:24]: Maybe say that this is for RPATH to work correctly, and refer to this PR. Without such comments, it's a bit unclear why you'd want to do this.

Comment (branfosj) [reply easybuild/easyconfigs/m/MAGeCK/MAGeCK-0.5.9.4-foss-2021a.eb]: bf90f29

**Comment by casparvl** [comment]
Test report by @casparvl
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
software1.lisa.surfsara.nl - Linux debian 10.11, x86_64, Intel(R) Xeon(R) Silver 4110 CPU @ 2.10GHz, Python 2.7.16
See https://gist.github.com/621d3bfd222a9d55d824423b27534309 for a full test report.

**Review by casparvl — APPROVED** [review]
Thanks @branfosj , looks good to me now! :)

**Comment by casparvl** [comment]
Going in, thanks @branfosj!

**Pull request merged**